src/xunit.netcore.extensions/Discoverers/ConditionalTestDiscoverer.cs

```diff
@@ -48,16 +48,36 @@ internal class ConditionalTestDiscoverer
                     continue;
                 }
 
-                string[] symbols = conditionMemberName.Split('.');
                 Type declaringType = testMethodDeclaringType;
 
-                if (symbols.Length == 2)
+                // We have qualified type name with the assembly name, something like
+                // [ConditionalFact("System.PlatformDetection, CoreFx.Private.TestUtilities!" + nameof(PlatformDetection.IsNonZeroLowerBoundArraySupported))]
```

Comment:
I just missed that the comma was inside quotes :( so that is why I thought it was wrong, I thought you where passing 2 parameters in the example. Sorry about that. 

If the qualified name is expected as you suggest and it is easier I'm fine with sticking with that, however I think the other one is much clearer. It is unfortunate that it the qualified name structure.

Thanks for clarifying @tarekgh